src/game/server/bot.cpp

```diff
@@ -341,8 +341,21 @@ void CBot::Tick()
 
 	m_RealTarget = m_Target + Pos;
 
-	if(g_Config.m_SvBotAllowFire && m_pPlayer->GetCharacter()->CanFire())
-		HandleWeapon(InSight);
+	if(g_Config.m_SvBotAllowFire && m_pPlayer->GetCharacter()->botPreFire(g_Config.m_SvBotDelay))
```

Comment:
This needs to handle CanFire and BotPreFire. CanFire is to check if it's possible to shoot at all. There might be a situation where enough time has passed to satisfy the bot delay, but not enough time to satisfy reload delay, so these shouldn't be mixed together.

Reply:
Like explained in the other conversation, it's better for HandleWeapon to be executed before the reloading is done to not extra delay to the shooting only the reactiontime of the bot. The actual shooting happens in ShootWeapon, which only gets executed when CanFire() would also return true.